### PR TITLE
Update android to NDK 15c

### PR DESCRIPTION
**Pull request by HalfdanJ**
Updated the NDK to 15c, and fixed all the issues it created.

### CODE REVIEW

File: apothecary/formulas/FreeImage/FreeImage.sh [hunk 1]
```diff
@@ -61,13 +62,14 @@ function prepare() {
 	    cat > Source/LibRawLite/src/swab.h << ENDDELIM
 	    #include <stdint.h>
         #include <asm/byteorder.h>
-        inline void swab(const void *from, void*to, ssize_t n)
+		#define ___swab(x)  ({ __u16 __x = (x);   ((__u16)(   (((__u16)(__x) & (__u16)0x00ffU) << 8) | (((__u16)(__x) & (__u16)0xff00U) >> 8) ));  })
```
Comment (HalfdanJ) [inline apothecary/formulas/FreeImage/FreeImage.sh]: __arch_swab16  no longer works, i found a different repo that used this as a workaround, but not sure if it's correct.

**Comment by arturoc** [comment]
i left the minimum api at 19 cause bumping it up made the binaries not compatible with 19 or less not sure how things are right now but is it possible to keep compatibility with at least 19?

**Comment by HalfdanJ** [comment]
Yes, bringing up the target should not break the minimum. Only if you start using features implemented later. This is how it's also setup if you create a new Java app, and in unity. 

The benefit of bringing the sdk up is it's then possible to make apps that depends on higher sdk, for example tango.

**Comment by arturoc** [comment]
yes supposedly it should work like that but last time i bumped it up
over 19 binaries wouldn't work on phones that had 19 or older. there's
some symbols in the standard library that are not present in 19 or less
that are in 20 or more which gives undefined symbol errors when trying
to run in those phones.

I'm talking about the ndk only

**Comment by HalfdanJ** [comment]
Ok, I'll do some digging and try it on some test phones at work.

**Comment by HalfdanJ** [comment]
Damn, You are totally right arturo. Here is the holy grail that I just found on this issue: https://stackoverflow.com/a/41079462/300830 

The tldr; is that the API level NDK is compiling with has to be equal to the MIN_API_LEVEL. 

I just tried to lower the api level to 19, and the build is of course completely broken then....

**Comment by HalfdanJ** [comment]
Moved to new pr's

**Pull request closed without merging**